src/server/index.js

```diff
@@ -9,11 +9,124 @@
 const Koa = require('koa');
 const IO = require('koa-socket');
 const Redux = require('redux');
+import Crypto from 'crypto';
+import Router from 'koa-router';
+import koaBody from 'koa-body';
 import { InMemory, Mongo } from './db';
 import { createGameReducer } from '../core/reducer';
+
 const PING_TIMEOUT = 20 * 1e3;
 const PING_INTERVAL = 10 * 1e3;
 
+const isActionFromAuthenticPlayer = async ({
+  action,
+  db,
+  gameID,
+  playerID,
+}) => {
+  const gameMetadata = await db.get(`${gameID}:metadata`);
+  if (!gameMetadata) {
+    return true;
+  }
+
+  const hasSecrets = Object.keys(gameMetadata.players).some(key => {
+    return !!gameMetadata.players[key].secret;
+  });
+  if (!hasSecrets) {
+    return true;
+  }
+
+  if (!action.payload.secret) {
+    return false;
+  }
+
+  if (action.payload.secret !== gameMetadata.players[playerID].secret) {
+    return false;
+  }
+
+  return true;
+};
+
+const createSecret = async () => {
+  return new Promise((resolve, reject) => {
+    Crypto.randomBytes(32, (err, buf) => {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(buf.toString());
+      }
+    });
+  });
+};
+
+const getNewGameInstanceID = ({ gameName }) => {
+  // TODO: Return a new instance ID for this game
```

Comment:
This needs to generate a random instance ID. UUID generation would probably be sufficient, if our gameIDs are strings throughout.